### PR TITLE
No-batch-dim support for ConvTransposeNd

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70507
* #70506

This PR adds no-batch-dim support for:
* `nn.ConvTranspose1d`, `nn.ConvTranspose2d`, `nn.ConvTranspose3d`
* `nn.LazyConvTranspose1d`, `nn.LazyConvTranspose2d`, `nn.LazyConvTranspose3d`
* `F.conv_transpose1d`, `F.conv_transpose2d`, `F.conv_transpose3d`

The support is added at the op level (e.g. `conv_transpose1d`) and tested via `ModuleInfo` entries with a reference function that compares the output from a single batch item input vs. the output from a no-batch-dim input.

Differential Revision: [D33355035](https://our.internmc.facebook.com/intern/diff/D33355035)

### CODE REVIEW

**Comment by jbschlosser** [comment]
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33355035).

**Comment by jbschlosser** [comment]
@jbschlosser has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33355035).

**Comment by jbschlosser** [comment]
Closing since splitting didn't work as hoped
